### DISCUSSION TITLE
Does velox support hudi now ?

### DISCUSSION THREAD

**Discussion by jiangjiguang** [facebookincubator/velox#4088] [General]
Does velox support data lake hudi now ？ if no, community has the roadmap when to support hudi ? thanks.

**Comment by pedroerp** [comment]
Not yet. This half we are improving the reader to add support for merge-on-read and other APIs which will be used for Hudi support (@Yuhta will be working on this). 

Ahana has Hudi support in their radar, but I'm not sure what the timeline would look like (cc: @majetideepak @aditi-pandit).

Alternatively, would you like to help us add support for it?

**Reply by Yuhta** [reply]
We probably need to have the interface ready before anyone from the community working on the actual implementation of Hudi mutability.  We will have the interface ready in 2023 H1.
